### PR TITLE
driver/net/lan9250: Add lan9250_ioctl and lan9250_uninitialize APIs

### PR DESCRIPTION
**Pull request by cwespressif**
## Summary

- Add lan9250_ioctl and lan9250_uninitialize APIs

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by jerpelea — APPROVED** [review]
please add ad commit message

**Comment by acassis** [comment]
Hi Chen Wen @cwespressif , nice you see your around again, please do a rebase to try fix the CI error

**Comment by cwespressif** [comment]
> please add ad commit message

Done

**Comment by cwespressif** [comment]
> fix the CI error

Hi @acassis , nice to meet you too, it seems to be a download problem:

> fatal: unable to access 'https://github.com/espressif/esp-hal-3rdparty.git/': Failed to connect to github.com port 443 after 133784 ms: Connection timed out

I will rebase to see if this can fix the CI error, thanks.

File: drivers/net/lan9250.c [hunk 1]
```diff
@@ -2322,6 +2327,79 @@ static int lan9250_rmmac(FAR struct net_driver_s *dev,
...
+
+  /* Decode and dispatch the driver-specific IOCTL command */
+
+  switch (cmd)
+    {
+#ifdef CONFIG_NETDEV_PHY_IOCTL
+      case SIOCGMIIPHY: /* Get MII PHY address */
+        {
+          struct mii_ioctl_data_s *req =
+                    (struct mii_ioctl_data_s *)((uintptr_t)arg);
```
Comment (pkarashchenko) [inline drivers/net/lan9250.c:2369]: `FAR` missing here and in other places of switch/case

File: drivers/net/lan9250.c [hunk 2]
```diff
@@ -2322,6 +2327,79 @@ static int lan9250_rmmac(FAR struct net_driver_s *dev,
...
+          lan9250_set_phyreg(priv, req->reg_num, req->val_in);
+        }
+        break;
+#endif /* CONFIG_NETDEV_PHY_IOCTL */
+
+    default:
+      nerr("ERROR: Unrecognized IOCTL command: %d\n", cmd);
+      ret = -ENOTTY; /* Special return value for this case */
```
Comment (pkarashchenko) [inline drivers/net/lan9250.c:2393]: Needs 2 spaces more for indent

**Pull request merged**